posthog/celery.py

```diff
@@ -108,6 +111,12 @@ def setup_periodic_tasks(sender: Celery, **kwargs):
 
     sender.add_periodic_task(120, calculate_cohort.s(), name="recalculate cohorts")
 
+    sender.add_periodic_task(
+        crontab(hour=0, minute=randrange(0, 40)),  # every day at a random minute past midnight.
```

Comment:
We can end up with multiple instances of the task running due to deploys with this approach - let's not try be clever?